jina/types/arrays/mixins/reduce.py

```diff
@@ -0,0 +1,47 @@
+from typing import List, TYPE_CHECKING
+
+
+if TYPE_CHECKING:
+    from ..document import DocumentArray
+    from ...document import Document
+    from ....helper import T
+
+
+class ReduceMixin:
+    """A mixing that provides reducing logic for :class:`DocumentArray` or :class:`DocumentArrayMemmap`"""
+
+    def reduce(self: 'T', da: 'T'):
+        """
+        Reduces da and the current DocumentArray into one DocumentArray in-place. Changes are applied to the current
+        DocumentArray
+        :param da: DocumentArray
+        """
+        for doc in da:
+            if doc.id in self:
+                self._reduce_doc(self[doc.id], doc)
+            else:
+                self.append(doc)
+
+    @staticmethod
+    def _reduce_doc(doc1: 'Document', doc2: 'Document'):
+        """
+        Reduces doc1 and doc2 into one Document in-place. Changes are applied to doc1
+        :param doc1: first Document
+        :param doc2: second Document
+        """
+        if len(doc2.matches) > 0:
+            doc1.matches.reduce(doc2.matches)
+
+        if len(doc2.chunks) > 0:
+            doc1.chunks.reduce(doc2.chunks)
```

Comment:
so we still dont have a conclusion on how to reduce on other properties right? what about following the logic of `MergeFrom`: https://googleapis.dev/python/protobuf/latest/google/protobuf/message.html#google.protobuf.message.Message.MergeFrom

> Singular fields that are set in the specified message overwrite the corresponding fields in the current message. Repeated fields are appended. Singular sub-messages and groups are recursively merged.

Reply:
I suggest the following logic: all fields that are set except matches and chunks overwrite the corresponding fields in the current document.
Content attributes (text, blob, ...) are ignored and only `content` is considered

Reply:
I applied the logic from MergeFrom using the method `update`
However, after some thoughts, I think it's better to always keep the attributes that are set on the left-most document.
This will avoid redundantly setting the attributes each time we reduce and will save time
@JoanFM your thoughts ?

Reply:
reduction on property attributes is applied with priority to the left-most document arrays